examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,504 @@
+import typer
+import json
+import logging
+from rich.prompt import Prompt, Confirm
+from typing import List, Tuple, Optional, Literal, Any
+
+import langroid as lr
+from langroid.language_models import OpenAIGPTConfig
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.agent.task import Task
+from langroid.agent.tools.metaphor_search_tool import MetaphorSearchTool
+from langroid.utils.logging import setup_logger
+from langroid.utils.configuration import set_global
+from langroid.agent.tools.orchestration import DoneTool
+
+from config import get_base_llm_config, get_global_settings
+from models import SystemMessages, Message
+
+DEFAULT_SYSTEM_MESSAGE_ADDITION = f"""
+            DO NOT REPEAT ARGUMENTS THAT HAVE BEEN PREVIOUSLY GENERATED 
+            AND CAN BE SEEN IN THE DEBATE HISTORY PROVIDED. 
+            """
+FEEDBACK_AGENT_SYSTEM_MESSAGE = f"""  
+            STEP 1: Provide Thorough Debate Feedback
+            You are an expert and experienced judge specializing in Lincoln-Douglas style debates. 
+            Your goal is to evaluate the debate thoroughly based on the following criteria:
+            1. Clash of Values: Assess how well each side upholds their stated value (e.g., justice, morality) 
+               and how effectively they compare and prioritize values.
+            2. Argumentation: Evaluate the clarity, organization, and logical soundness of each side's case structure, 
+               contentions, and supporting evidence.
+            3. Cross-Examination: Judge the effectiveness of questioning and answering during cross-examination.
+            4. Rebuttals: Analyze how well each side refutes their opponent's arguments.
+            5. Persuasion: Assess communication quality, tone, rhetorical effectiveness, and emotional/ethical appeals.
+            6. Technical Execution: Identify if major arguments were addressed or dropped and check consistency.
+            7. Debate Etiquette: Evaluate professionalism, respect, and demeanor.
+            8. Final Focus: Judge the strength of closing speeches, how well they summarize the case, 
+            and justify a winner.
+            Provide constructive feedback for each debater, 
+            summarizing their performance and declaring a winner with justification.   
+            """
+METAPHOR_SEARCH_AGENT_SYSTEM_MESSAGE_TEMPLATE = """
```

Comment:
I don't think this is giving sufficient context to the metaphor agent -- we need a preamble stating the background, i.e. that the task is to find good references for arguing two sides of a debate.

Reply:
done